vllm/config.py

```diff
@@ -350,60 +364,90 @@ def _verify_tokenizer_mode(self) -> None:
                 "either 'auto', 'slow' or 'mistral'.")
         self.tokenizer_mode = tokenizer_mode
 
+    def _get_preferred_task(
+        self,
+        architectures: List[str],
+        supported_tasks: Set[_ResolvedTask],
+    ) -> Optional[_ResolvedTask]:
+        model_id = self.model
+        if get_pooling_config(model_id, self.revision):
+            return "embed"
+        if model_id.startswith("cross-encoder/") or "-reranker" in model_id:
+            return "score"
```

Comment:
I think this won't work if user specifies a custom local model path instead of hf repo_id.

Can we check if the target model is a cross-encoder to prefer `score` task?

Reply:
Oh, completely forgot about that! That should work, thanks!